src/Exception/CouldNotTransitionException.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Dhii\Data\Exception;
+
+use Dhii\Data\StateAwareInterface;
+use Dhii\Data\StateAwareTrait;
+use Dhii\Data\TransitionerInterface;
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Util\Normalization\NormalizeStringableCapableTrait;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * An exception thrown when a transitioner fails to transition.
+ *
+ * @since [*next-version*]
+ */
+class CouldNotTransitionException extends AbstractBaseTransitionerException implements
+    CouldNotTransitionExceptionInterface
+{
+    /* @since [*next-version*] */
+    use NormalizeStringableCapableTrait;
+
+    /**
+     * The transition, if any.
+     *
+     * @since [*next-version*]
+     *
+     * @var string|Stringable|null
+     */
+    protected $transition;
+
+    /**
+     * The subject, if any.
+     *
+     * @since [*next-version*]
+     *
+     * @var StateAwareInterface|null
+     */
+    protected $subject;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null     $message      The error message, if any.
+     * @param int|null                   $code         The error code, if any.
+     * @param RootException|null         $previous     The previous exception for chaining, if any.
+     * @param TransitionerInterface|null $transitioner The transitioner, if any.
+     * @param StateAwareInterface|null   $subject      The subject, if any.
+     * @param string|Stringable|null     $transition   The transition, if any.
+     */
+    public function __construct(
+        $message = null,
+        $code = null,
+        $previous = null,
+        TransitionerInterface $transitioner = null,
+        $subject = null,
+        $transition = null
+    ) {
+        $this->_initBaseException($message, $code, $previous);
+        $this->_setTransitioner($transitioner);
+        $this->_setTransition($transition);
+        $this->_setSubject($subject);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getTransition()
+    {
+        return $this->transition;
+    }
+
+    /**
+     * Sets the transition.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null $transition The transition, if any.
+     *
+     * @throws InvalidArgumentException If the argument is not a string or stringable object.
+     */
+    protected function _setTransition($transition)
```

Comment:
Looks like awareness can very easily be extracted into traits and put into [dhii/data-state-abstract][].

[dhii/data-state-abstract]: https://github.com/Dhii/data-state-abstract